scipy/sparse/linalg/_norm.py

```diff
@@ -0,0 +1,174 @@
+    
+    The following norms can be calculated:
+    
+    =====  ============================  
+    ord    norm for sparse matrices             
+    =====  ============================  
+    None   Frobenius norm                
+    'fro'  Frobenius norm                
+    inf    max(sum(abs(x), axis=1))      
+    -inf   min(sum(abs(x), axis=1))      
+    0      abs(x).sum(axis=axis)                           
+    1      max(sum(abs(x), axis=0))      
+    -1     min(sum(abs(x), axis=0))      
+    2      Not implemented  
+    -2     Not implemented      
+    other  Not implemented                               
+    =====  ============================  
+    
+    The Frobenius norm is given by [1]_:
+    
+        :math:`||A||_F = [\\sum_{i,j} abs(a_{i,j})^2]^{1/2}`
+    
+    References
+    ----------
+    .. [1] G. H. Golub and C. F. Van Loan, *Matrix Computations*,
+        Baltimore, MD, Johns Hopkins University Press, 1985, pg. 15
+    
+    Examples
+    --------
+    >>> from scipy.sparse import *
+    >>> import numpy as np
+    >>> from scipy.sparse.linalg import norm
+    >>> a = np.arange(9) - 4
+    >>> a
+    array([-4, -3, -2, -1, 0, 1, 2, 3, 4])
+    >>> b = a.reshape((3, 3))
+    >>> b
+    array([[-4, -3, -2],
+           [-1, 0, 1],
+           [ 2, 3, 4]])
+           
+    >>> b = csr_matrix(b)
+    >>> norm(b)
+    7.745966692414834
+    >>> norm(b, 'fro')
+    7.745966692414834
+    >>> norm(b, np.inf)
+    9
+    >>> norm(b, -np.inf)
+    2
+    >>> norm(b, 1)
+    7
+    >>> norm(b, -1)
+    6
+    
+    Using the `axis` argument to compute vector norms:
+    
+    >>> c = np.array([[ 1, 2, 3],
+    ...               [-1, 1, 4]])
+    >>> c = csr_matrix(c)
+    >>> norm(c, axis=0)
+    matrix[[ 1.41421356, 2.23606798, 5. ]]
+    >>> norm(c, axis=1)
+    matrix[[ 3.74165739, 4.24264069]]
+    >>> norm(c, ord=1, axis=1)
+    matrix[[6]
+           [6]]
+
+"""
+    if not issparse(x):
+        raise TypeError("input is not sparse. use numpy.linalg.norm")
+
+    # Check the default case first and handle it immediately.
+    if ord in [None, 'fro', 'f'] and axis is None:
+        if isComplexType(x.dtype.type):
+            sqnorm = dot(x.real, x.real) + dot(x.imag, x.imag)
+        else:
+            sqnorm = x.power(2).sum()
+        return sqrt(sqnorm)
+
+    # Normalize the `axis` argument to a tuple.
+    nd = x.ndim
+    if axis is None:
+        axis = tuple(range(nd))
+    
+    if np.isscalar(axis):
+        if ord == Inf:
+            return max(abs(x).sum(axis=axis))
+        elif ord == -Inf:
+            return min(abs(x).sum(axis=axis))
+        elif ord == 0:
+            # Zero norm
+            return (x != 0).sum(axis=axis)
+        elif ord == 1:
+            # special case for speedup
+            return abs(x).sum(axis=axis)
+        elif ord == -1:
+            return min(abs(x).sum(axis=axis))             
+        elif ord is None:            
```

Comment:
Following the numpy norm convention, I think this is correct as-is.  When an axis is specified, the vector norm instead of the matrix norm is used.  According to numpy's conventions, explicitly requesting the Frobenius norm of a vector is not allowed -- it is only allowed for a matrix.  On the other hand, the default norm for both matrices and vectors is "square root of sums of squared magnitudes of entries" which does correspond to the Frobenius norm of a matrix.